tests/api_tests/nodes/test_views.py

```diff
@@ -879,6 +879,514 @@ def test_filtering_node_with_non_bibliographic_contributor(self):
+        assert_equal(res.status_code, 400)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user), ['read', 'write', 'admin'])
+
+    def test_remove_all_bibliographic_statuses_contributors(self):
+        self.project.set_visible(self.user_two, False, save=True)
+        self.project.reload()
+        
+        data = {
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_creator, data, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+        self.project.reload()
+        assert_true(self.project.get_visible(self.user))
+
+    def test_change_contributor_non_admin_auth(self):
+        data = {
+            'permission': 'read',
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user_two.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), ['read', 'write'])
+        assert_true(self.project.get_visible(self.user_two))
+
+    def test_change_contributor_not_logged_in(self):
+        data = {
+            'permission': 'read',
+            'bibliographic': False
+        }
+        res = self.app.put_json(self.url_contributor, data, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), ['read', 'write'])
+        assert_true(self.project.get_visible(self.user_two))
+
+
+class TestNodeContributorDelete(ApiTestCase):
+    def setUp(self):
+        super(TestNodeContributorDelete, self).setUp()
+        self.user = AuthUserFactory()
+        self.user_two = AuthUserFactory()
+        self.user_three = AuthUserFactory()
+
+        self.project = ProjectFactory(creator=self.user)
+        self.project.add_contributor(self.user_two, permissions=['read', 'write'], visible=True, save=True)
+        self.project.reload()
+
+        self.url_user = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user._id)
+        self.url_user_two = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user_two._id)
+        self.url_user_three = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user_three._id)
+
+    def test_remove_contributor_admin(self):
+        res = self.app.delete(self.url_user_two, auth=self.user.auth)
+        assert_equal(res.status_code, 204)
+
+        self.project.reload()
+        assert_not_in(self.user_two, self.project.contributors)
+
+    def test_remove_contributor_non_admin(self):
```

Comment:
Despite the test name, neither user 2 nor 3 appear to be admins. Both are added with r+w permissions.

A better name might be `test_non_admin_contributor_cannot_delete_other_contributors`.

(And an even **better** name would be more concise...!)